### PR TITLE
libical(-devel): upgrade to version 2.0.0(+)

### PR DESCRIPTION
**Pull request by mkae**
- use cmake PG 1.1
- introduce a docs variant
- removed invalid keyword 'Id'
- remove unneeded patches

### CODE REVIEW

File: devel/libical/files/patch-nodocs.diff [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 23898b6..fb7e452 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -347,7 +347,7 @@ endif()
+ ################# build subdirs ########################
+ 
+ add_subdirectory(design-data)
+-add_subdirectory(doc)
++# add_subdirectory(doc)
```
Comment (mojca) [inline devel/libical/files/patch-nodocs.diff]: Not saying that you should wait for upstream to resolve the issue, but if you have the energy to come up with a proper patch or at least submit a feature request upstream that implements a configure option to switch building the docs on and off, that would be helpful.

File: devel/libical/Portfile [hunk 2]
```diff
@@ -1,22 +1,19 @@
...
 
-cmake.out_of_source yes
-
 name                libical
 
 if {${subport} eq "${name}"} {
-	github.setup    libical libical 1.0.1 v
-	conflicts       libical-devel
+    github.setup    libical libical 2.0.0 v
+    conflicts       libical-devel
```
Comment (mojca) [inline devel/libical/Portfile]: Could you make a separate commit for whitespace-only (or any other non-funtional) changes?

File: devel/libical/Portfile [hunk 3]
```diff
@@ -35,22 +32,33 @@ subport libical-devel {
...
-    checksums       rmd160  780f9fb08b324536af362f8812ff671e4d1082d0 \
-                    sha256  b3898c9ca3522ba4f0454e8b7c26a85b2b1259e16d670cccf7abb035c8a6caa9
+    # v 2.0.0
+    checksums       rmd160  1901573af1fb57be8feca3b3dd9db758c3e136ee \
+                    sha256  c9c076153867cd6ebac23febb276036597443eabd30ba4534d38950a2cbb889e
 } else {
-    # v1.0.1-351-g25bc500
-	checksums       rmd160  4379c5d6d31a78ecb2ceaff6e64a8900e9b30cfe \
-	                sha256  f7c22344a461f8ee1ce65e3efa477e9846b1ca39f44af622a149ce37f3850332
+    # v2.0.0-115-g1b83385
```
Comment (mojca) [inline devel/libical/Portfile]: A matter of personal preference and completely orthogonal to your changes. I would hate to write the version at several different places as I would probably forget to change it next time. You could also put the checksums right below the version definition.

**Comment by dbevans** [comment]
Since this is a major release, before merging, please make sure that no dependent ports will be broken due to API incompatibilities, etc.  Thanks.

**Comment by mojca** [comment]
> Since this is a major release, before merging, please make sure that no dependent ports will be broken due to API incompatibilities, etc.

I guess that the pull request should also include a revbump of all the dependents (I only found `gnokii`, `evolution-data-server`, `gnome-calendar`, `kdepimlibs4`).

File: devel/libical/Portfile (hunk 2 above)
Comment (mkae) [inline devel/libical/Portfile]: Is this really a must? This affects only a handful of lines for this port... The required `git`-fu to insert an additional changeset just for this seems like overkill to me right now. :-)

File: devel/libical/Portfile (hunk 3 above)
Comment (mkae) [inline devel/libical/Portfile]: @RJVB, what do you think about this?

**Comment by mkae** [comment]
I am not using gnome, so it will be hard for me to test the first 3 ports. @dbevans, you're maintaining `evolution-data-server` and `gnome-calendar`, can you test it with this new `libical` version?

I'm going to contact `gnokii`'s maintainer.

@RJVB stated that there are no negative effects to be seen for `kdepimlibs4`.

File: devel/libical/files/patch-nodocs.diff (hunk 1 above)
Comment (RJVB) [inline devel/libical/files/patch-nodocs.diff]: From KDE's extra cmake modules:

```
# ECMOptionalAddSubdirectory
# --------------------------
#
# Make subdirectories optional.
#
# ::
#
#   ecm_optional_add_subdirectory(<dir>)
#
# This behaves like add_subdirectory(), except that it does not complain if the
# directory does not exist.  Additionally, if the directory does exist, it
# creates an option to allow the user to skip it. (-DBUILD_<dir>).
```

This will be easier to propose upstream for KDE projects, but in case of libical it adds a new dependency. In all cases it will depend a lot on upstream goodwill, esp. with KDE where "doc" subdirectories usually build the project's "handbook" which is available through a menu (and that menu will be there regardless of whether we build the handbook).

Comment (mkae) [inline devel/libical/files/patch-nodocs.diff]: Well, as this isn't a KDE project I could imagine that upstream probably won't want ECM as an additional dependency.

Anyway, what would one have to add to the `cmake` options? Would it be `-DBUILD_doc=FALSE`

**Comment by RJVB** [comment]
The 2.0.0 release notes do contain a warning:

```
WARNING: Version 2 IS NOT Binary Compatible with Older Versions
Version 2 is Source Compatible with Older Versions
```

It's true that I haven't seen problems in the code I run that uses libical (KDE PIM4), but I must be honest and admit that I cannot remember if I rebuilt or not (probably a minor partial rebuild since I keep the build directories for kdepimlibs4 around).

**Comment by mojca** [comment]
> I'm going to contact gnokii's maintainer.

Good luck. It looks more like something ready for a port abandoned ticket.

In any case you'll probably need a commit revbumping dependencies.

File: devel/libical/Portfile (hunk 2 above)
Comment (mojca) [inline devel/libical/Portfile]: It's not a must, but I did it for you.

Comment (mkae) [inline devel/libical/Portfile]: Thanks, Mojca! Doing it directly in the current version is something I should have been able to come up with myself. So super easy. Was blind.

File: devel/libical/Portfile (hunk 3 above)
Comment (mkae) [inline devel/libical/Portfile]: Sounds like a good idea.

File: devel/libical/files/patch-nodocs.diff (hunk 1 above)
Comment (mojca) [inline devel/libical/files/patch-nodocs.diff]: I would go for capital letters, but that's a matter of personal preference

```
-DBUILD_DOC[S]=FALSE
```

Again, there's nothing wrong if you don't change this, it's just a tiny bit cleaner to use a configure option and install a patch unconditionally (or even better to convince upstream to include it).

File: devel/libical/Portfile (hunk 2 above)
Comment (mojca) [inline devel/libical/Portfile]: There is no difference whether you do it to master or to another branch, you need to "rebase" or reapply the same changes in any case. But I decided to commit to master because the changes were straightforward and there was no need to keep them in this PR for discussion, it makes the PR cleaner this way.

Comment (mkae) [inline devel/libical/Portfile]: Agreed. :+1

File: devel/libical/files/patch-nodocs.diff (hunk 1 above)
Comment (RJVB) [inline devel/libical/files/patch-nodocs.diff]: Yes, with this particular solution one would have to add `-DBUILD_doc=FALSE` (or OFF). Presuming CMake options are case-sensitive it cannot be all-capital because the part after the underscore reflects the name of the directory.

File: devel/libical/Portfile (hunk 3 above)
Comment (RJVB) [inline devel/libical/Portfile]: What, upgrading? You'll need to leave me some time for that, I'm swamped ATM..

**Comment by mkae** [comment]
Yes, @mojca, I'll add such a revbump commit and fix this PR.

**Comment by mkae** [comment]
@dbevans, you're maintaining `evolution-data-server` and `gnome-calendar`, can you test it with this new libical version?

**Comment by RJVB** [comment]
Let me rephrase what I said earlier. The version upgrade appears to require a revbump, at least for kdepimlibs4. One of my other dependent ports (KF5 kcalcore) seems to work fine after upgrading from 2.0.0.115 to 2.0.0.249 just now, but KMail stopped working.

**Comment by mkae** [comment]
> KMail stopped working

KDE4's KMail?

File: devel/libical/files/patch-nodocs.diff (hunk 1 above)
Comment (mkae) [inline devel/libical/files/patch-nodocs.diff]: @RJVB, will you contact upstream regarding them making use of `ECM` or finding another approach to the doc avoiding?

**Comment by RJVB** [comment]
On Thursday January 12 2017 15:26:48 Marko K�ning wrote:
> > KMail stopped working
> 
> KDE4's KMail?

yes. kdepimlibs4 needed a rebuild.

**Comment by mkae** [comment]
Ah, ok, that means that it would work again if `kdepimlibs4` got rebuilt. Well, that is why my rev-bump commit is part of this PR.

**Comment by kencu** [comment]
Is this PR still on life support after all these months, or should it be closed and revisited at some later time? Looks like MacPorts has moved past this point in time to me...

**Comment by raimue** [comment]
I assume this was superseded by e30a94e2a2b250fc478ca2c7c68bca10ac300189 a while ago. Closing.
